[info arch] Add filters to exampleIndex layout (#345)
mapbox/dr-ui: docs/src/pages/guides/batfish-helpers.md

```js
import React from 'react';
import filters from '@mapbox/batfish/data/filters.js';

class PageShell extends React.Component {
  render() {
    return (
      <PageLayout filters={filters} />;
    )
  }
}
```

### Output

- The shape of filters is an object, where the top-level keys are pathnames for `examplesIndex` pages.
  - Each object has a `topics` array, a unique list of topics.
  - Each objects has `pages` array of all sub pages to display. The `pages` array is ordered by the `order` frontMatter property and then alphabetically by title.
  - If available, the object may have a `levels` array, a unique list of levels options.
  - If available, the object may have a `languages` array, a unique list of language options.

>   - If available, the object may have a `videos` array, an array of all pages with `video: true` in the frontMatter.

Reviewer: Note: the next section "Append topics helper" will be deleted in #347 so I didn't update old references to topics here.

>   - If available, the object may have a `videos` array, an array of all pages with `video: true` in the frontMatter.

Reviewer: ```suggestion
  - If available, the object may have `videos: true` to indicate that some of its sub pages have `video: true` in the frontMatter.
```

```js
import React from 'react';
import filters from '@mapbox/batfish/data/filters.js';

class PageShell extends React.Component {
  render() {
    return (
      <PageLayout filters={filters} />;
    )
  }
}
```

### Output

- The shape of filters is an object, where the top-level keys are pathnames for `examplesIndex` pages.
  - Each object has a `products` array, a unique list of products.
  - Each object has a `topics` array, a unique list of topics.

>   - Each object has a `pages` array of all sub pages to display. The `pages` array is ordered by the `order` frontMatter property and then alphabetically by `title`.

Reviewer 2: Is this still true given the `pageSorter` function in the filter helper?

Reviewer: It's not, I'll update!
